README.md

```diff
@@ -1,2 +1,145 @@
+
+_.c2crc_
+```json
+{
+  "consul": {
+    "host": "localhost",
+    "port": 8500
+  },
+  "[..]": "[..]"
+}
+```
+
+#### Docker
+
+The `docker` section of the configuration is given to [dockerode](https://github.com/apocas/dockerode) constructor.
+
+Please refer to [dockerode](https://github.com/apocas/dockerode) documentation for supported options.
+
+Default configuration listens to Docker daemon on `/var/run/docker.sock` unix socket.
+
+_.c2crc_
+```json
+{
+  "[..]": "[..]",
+  "docker": {
+    "socketPath": "/var/run/docker.sock"
+  }
+}
+```
+
+#### Logger
+
+Container-to-Consul uses [bunyan](https://github.com/trentm/node-bunyan). The `logger` section of the configuraton object is passed to `bunyan.createLogger` method.
+
+Default configuration outputs logs at `info` level to `stdout`.
+
+_.c2crc_
+```json
+{
+  "[..]": "[..]",
+  "logger": {
+    "name": "container-to-consul"
+  }
+}
+```
+
+
+## Container registration
+
+Container-to-Consul relies on [Docker labels](https://docs.docker.com/engine/userguide/labels-custom-metadata/) to define the values sent to Consul.
+
+Internally, Container-to-Consul calls Consul low-level [catalog/register](https://www.consul.io/docs/agent/http/catalog.html#catalog_register) and [catalog/deregister](https://www.consul.io/docs/agent/http/catalog.html#catalog_deregister) API endpoints.
+
+Container registration is an opt-in process. By default, Container-to-Consul will not register any container in Consul. To register a container the `consul.register` label must be set to `true`.
+
+### Supported labels
+
+| Label | Description | Default |
+| --- | --- | --- |
+| `consul.register` | Marks a container for discovery | `false` |
+| `consul.ip` | The IP that will be registered as part of the service in Consul| first IP found in the `NetworkSettings` group from `docker inspect` result. If the container is running in `host` network mode, the ip address used is the `docker0` one. |
+| `consul.port` | The port that the service will be discovered on| `none` |
+| `consul.service` | The name of the service in Consul | `container name` |
+| `consul.tags` | The tags that will be added to the service in Consul | `[]` |
+| `consul.kv.<KEY>` | Key/Value pairs | `{}` |
+
+### Registration
+Running;
+
+```
+docker run --name docker-nginx -d -l consul.register=true -l consul.ip=10.0.0.1 -l consul.service=nginx-service -l consul.tags=www,api -l consul.port=80 -l consul.kv.foo=bar -l consul.kv.bar=baz  nginx
+```
+
+
+will register a service `nginx-service` with
+- ip  `10.0.0.1`
+- tags `www` and `api`
+- port `80`
+- 2 key/value pairs under the `/services/nginx-service/` tree.
+
+You can check for the registered service in [Consul](http://localhost:8500/ui/dc1/services)
+
+> For a complete usage example please [see](https://github.com/mekomsolutions/container-to-consul/blob/main/e2e/docker/docker-compose.yaml).
+
+### Deregistration
+
+To deregister the container, you can simply stop it and remove it.
```

Comment:
Is removing the container necessary to deregister? I don't think so, right?
I suggest to then remove the `rm` command to put it a bit after, as a note or something.